PowerShell/ScubaGear/Testing/Unit/PowerShell/Providers/EXOProvider/Select-DohServer.Tests.ps1

```diff
@@ -0,0 +1,34 @@
+$ProviderPath = "../../../../../Modules/Providers"
+Import-Module (Join-Path -Path $PSScriptRoot -ChildPath "$($ProviderPath)/ExportEXOProvider.psm1") `
+    -Function 'Select-DohServer' -Force
+
+InModuleScope 'ExportEXOProvider' {
+    Describe -Tag 'ExportEXOProvider' -Name "Select-DohServer" {

```

Comment:
Should there be a test case to ensure that the logic to retrieve the 1.1.1.1 server works in case the other two fail dns servers fail?

Reply:
I considered doing that, and certainly still could add that, but at that point it felt more like I would be testing my mocked-up function, not the real function. I felt like having at least one case to make sure it would retry at least once was enough to make sure the mechanism worked. But I don't feel super strongly about that. One more nudge from you would be all I need to tip me to the other side.

Reply:
I wasn't sure what you mean by you would be testing the mocked-up function but use your judgement regarding whether any additional unit tests are needed. The more tests to ensure that the Select-DohServer works as expected, the better, if they add value.

Reply:
I'll add it. My concern was that as the complexity of the function I'm mocking approaches the complexity of the function I'm testing, the unit test will be more about testing the mock than the actual function. For example, this is what the mock function looks like for this case:
![image](https://github.com/cisagov/ScubaGear/assets/106177711/74c7cca3-6ecf-474a-b7bb-c2d96548a89e)

That said, after writing implementing it, I see it's really not that complex so I think this case is fine after all.

Reply:
Case in point, turns out that the mock I pasted above was buggy :joy:

I didn't know but looks like `Uri.ToString()` expands the IPv6 address to `[2606:4700:4700:0000:0000:0000:0000:1111]`. The actual function was fine but the unit test found a bug in my mock.